mrp_account_analytic/models/account_analytic_line.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    manufacturing_order_id = fields.Many2one(
+        "mrp.production",
+        string="Related Manufacturing Order",
+    )
+    stock_move_id = fields.Many2one(
```

Comment:
Hi Daniel. I would like to give you a couple of ideas that I think could help you to simplify the code a little bit. 
Since stock_move is related to mrp.production and mrp.workorder,  they could be taken from stock move in analytic line, directly
This is.. each stock_move (from order_id.raw_material_ids) is mandatory linked to an order_id and could be also linked to a workorder_id, so.. you would not need to create functions on each object. You could just link the stock_move to an analytic line and update values from him.

Fields order_id, workorder_id can not be directly related to stock_move, just because if you think about including workorders(operations) time cost in analytic on future, you will need both fields free to assign a value. 

But... I think both fields could be computed. Very easy. If analytic line is linked to stock, move then take from him. If it's linked to workorder.time_ids, take directly from him. 

I think this couple of changes could allow you to reduce the amount of code lines. Anyway, if you need me to test functionally the module, let me know!
Thank you for the contribution.
Ana

Reply:
Hello Ana, thank you for your feedback.
Indeed the `manufacturing_order_id` here could be computed from the `stock_move_id`or the `workorder_id`, instead on being directly set.

I felt the current approach would give more flexibility to developers, and changing it would remove only two lines of code, to be replaced by a computed method. I'm not sure the change would make the code simpler, but I'm open for more suggestions.

If you can test this, it would be great.
Thank you